src/components/ProductDetails.js

```diff
@@ -3,6 +3,113 @@ import '../App.css';
 import PropTypes from 'prop-types';
 
 class ProductDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      comment: '',
+      rating: 5,
+      feedbacks: [],
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleFeedback = this.handleFeedback.bind(this);
+    this.renderFeedbackSection = this.renderFeedbackSection.bind(this);
+    this.checkStorage = this.checkStorage.bind(this);
+    this.renderFeedbacks = this.renderFeedbacks.bind(this);
+  }
+
+  componentDidMount() {
+    this.checkStorage();
+  }
+
+  componentDidUpdate() {
+    const { email, comment, rating, feedbacks } = this.state;
+    const unsavedFeedback = {
+      email,
+      comment,
+      rating,
+    };
+    localStorage.setItem('unsavedFeedback', JSON.stringify(unsavedFeedback));
+    localStorage.setItem('feedbackList', JSON.stringify(feedbacks));
+  }
+
+  handleChange({ target }) {
+    this.setState({ [target.name]: target.value });
+  }
+
+  handleFeedback() {
+    const unsavedFeedback = JSON.parse(localStorage.getItem('unsavedFeedback'));
+    console.log(unsavedFeedback);
+    const { feedbacks } = this.state;
+    console.log(feedbacks);
+    this.setState({ feedbacks: [...feedbacks, unsavedFeedback] });
+  }
+
+  checkStorage() {
+    const feedbackList = localStorage.getItem('feedbackList');
+    const unsavedFeedback = localStorage.getItem('unsavedFeedback');
+    if (unsavedFeedback !== null) {
+      this.setState({
+        email: JSON.parse(unsavedFeedback).email,
+        comment: JSON.parse(unsavedFeedback).comment,
+        rating: JSON.parse(unsavedFeedback).rating,
+      });
+    }
+    if (JSON.parse(feedbackList) !== null) {
+      console.log(feedbackList);
+      this.setState({
+        feedbacks: JSON.parse(feedbackList),
+      });
+    }
+  }
+
+  renderFeedbacks(feedbacks) {
+    return (feedbacks.map((feedback) => (
+      <div key={ feedback.email } className="feedback">
+        <h4>{ feedback.rating }</h4>
+        <p>{ feedback.comment }</p>
+        <p>{ feedback.email }</p>
+      </div>
+    )));
+  }
+
+  renderFeedbackSection() {
+    const { email, comment, rating, feedbacks } = this.state;
+
+    return (
+      <form className="feedback-form">
+        <h4>Avaliações</h4>
+        <input
+          type="email"
+          name="email"
+          placeholder="ribeiro.brito@google.com"
```

Comment:
Como não temos uma label indicado o que é o campo, não seria melhor que o placeholder fizesse isso? Então ao invés de passarmos um e-mail, colocarmos a palavra e-mail? 
```suggestion
          placeholder="Email"
```